src/Emogrifier.php

```diff
@@ -1404,18 +1404,24 @@ private function getCssSelectorPrecedence($selector)
         $selectorKey = md5($selector);
         if (!isset($this->caches[static::CACHE_KEY_SELECTOR][$selectorKey])) {
             $precedence = 0;
-            $value = 100;
-            // ids: worth 100, classes: worth 10, elements: worth 1
-            $search = ['\\#', '\\.', ''];
+            $value = 10000;
+            $search = [
+                // ids: worth 10000
+                '\\#',
+                // classes, attributes, pseudo-classes (not pseudo-elements) except `:not`: worth 100
+                '(?:\\.|\\[|(?<!:):(?!not\\())',
+                // elements (not attribute values or `:not`), pseudo-elements: worth 1
+                '(?:(?<![="\':\\w-])|::)'
+            ];
 
             foreach ($search as $s) {
                 if (trim($selector) === '') {
                     break;
                 }
                 $number = 0;
-                $selector = preg_replace('/' . $s . '\\w+/', '', $selector, -1, $number);
+                $selector = preg_replace('/' . $s . '[\\w-]++/', '', $selector, -1, $number);
```

Comment:
Is this related to this PR? And if it is not, could you please undo this change?

And if this actually fixes a but, I'd prefer this change to have its own PR (including changelog entry and unit tests).

Reply:
I've reverted this change, as none of the tests I added fail without it.  When removing matched parts, I wanted to see `:first-child` rather than just `:first` removed, but maybe that isn't necessary.  I'll see if I can come up with a valid test case that would require it...